filesystem/File.php

```diff
@@ -238,7 +238,7 @@ public function exists() {
 	 */
 	public static function find($filename) {
 		// Get the base file if $filename points to a resampled file
-		$filename = preg_replace('/_resampled\/[^-]+-/', '', $filename);
+		$filename = preg_replace('/_resampled\/(.+\/|[^-]+-)/', '', $filename);
```

Comment:
This isn't going to be backwards compatible with the old filename pattern. Can you make the additional capture optional?

Also be careful of the `.` regexp pattern. It'll match slashes as well (which you probably want to avoid). Try `[^\\/]` instead?

You've also got the same logic as in HtmlEditorField. You should put this into a new method in file to get original filename from resampled filename. Then you can add a unit test just for that method. :) (hint hint).